programs/loader-v4/src/lib.rs

```diff
@@ -279,7 +280,7 @@ pub fn process_instruction_deploy(
     };
     let executor = ProgramCacheEntry::new(
         &loader_v4::id(),
-        environments.program_runtime_v2.clone(),
+        environments.program_runtime_v1.clone(),
```

Comment:
Should we split this in its own PR?

Reply:
This should have been in #2750 already but I missed it. Only noticed as I was working on the tests in #2796.